lib/ddtrace/contrib/configuration/resolvers/regexp_resolver.rb

```diff
@@ -0,0 +1,36 @@
+require 'ddtrace/contrib/configuration/resolver'
+
+module Datadog
+  module Contrib
+    module Configuration
+      # Resolves a value to a configuration key
+      module Resolvers
+        # Matches strings against Regexps.
+        class RegexpResolver < Datadog::Contrib::Configuration::Resolver
+          def resolve(name)
+            # Co-erce to string
+            name = name.to_s
+
+            # Try to find a matching pattern
+            matching_pattern = patterns.find do |pattern|
+              pattern.is_a?(Regexp) ? pattern =~ name : pattern == name
```

Comment:
I have a suggestion we can discuss for this matching: how about using case equality `#===` here?
It supports regular expressions, simple string matches, and procs (`pattern === name`). The logic would look simpler too.

I think procs might be something users might be interested in using for matching patterns here.

Reply:
Okay, I'll give case equality a shot.

Using procs could be good, too, although there's a small risk that in allowing that could end up with users introducing checks that have a high performance cost (as this is run per span, effectively.) Benefits probably outweigh this risk though, so I think its a good suggestion.

Reply:
Case equality seems to work nicely; also implemented Proc support. Thanks for the suggestion!